docs/docs/reference/other-new-features/tasty-reflect.md

```diff
@@ -24,7 +24,7 @@ To provide reflection capabilities in macros we need to add an implicit paramete
 import scala.quoted._
 import scala.tasty._
 
-inline def natConst(x: Int): Int = ~natConstImpl('x)
+inline def natConst(x: => Int): Int = ~natConstImpl('(x))
```

Comment:
That syntax does not work on master for me. This one did:

```suggestion
inline def natConst(x: => Int): Int = ${natConstImpl('{x})}
```

Also I believe the correct approach is to use `xTree.underlyingArgument` or `xTree.underlying` instead of a by-name argument. In principle, I think the second one should allow us to make `val x = 1; natConst(x)` work, but it does not seem to, currently. Maybe @nicolasstucki can tell us why?

Reply:
The correct approach is to use by-name parameters. 

The `underlyingArgument` will give a view of the original argument by using its content will brake some soundness. For example using `underlyingArgument` in `val x = {println("do once"); 1}; natConst(x)` where `natConst` does not take a by-name parameter may execute several times `println("do once")`.

`xTree.underlying` currently doesn't do much more than `xTree.underlyingArgument`. This was a compromise that had to be done to be able to support whitebox macros. `xTree.underlying` will probably be removed.

Reply:
What? No, it won't be unsound. What this macro output is `n.toExpr`. No duplication of effects AFAICT.

We've discussed this before: using by-names as the standard way for macros to inspect their arguments will often cause unintended swallowing of effects.

Reply:
Also I'm interested about the compromise to support whitebox macros. Could you elaborate? (Perhaps not here in this issue.)

Reply:
To support white box macros we need to expand macros in typer. We cannot use `underlying` to access the typed tree of any definition because those definitions might still not be typed.